posttroll/tests/test_pubsub.py

```diff
@@ -314,6 +314,23 @@ def test_pub_minmax_port(self):
             self.assertEqual(res, port)
             break
 
+    @mock.patch("posttroll.publisher.get_context")
+    def test_bind_retries(self, get_context):
+        """Test that the destination bind is retried on failure."""
+        from zmq.error import ZMQError
+        from posttroll.publisher import Publish, BIND_RETRIES
+        context = mock.MagicMock()
+        context.bind.side_effect = ZMQError("mocked failure")
+        get_context.return_value.socket.return_value = context
+
+        try:
+            with Publish("test_bind_retries", port=50000) as pub:
+                pass
+            raise AssertionError("OSError not raised")
+        except OSError:
+            pass
```

Comment:
How about asserting the OSError is raised?

Reply:
eg with pytest:
```python
with pytest.raises(OSError):
    with Publish("test_bind_retries", port=50000):
        pass
```

Reply:
Done.